src/main/java/vimification/model/LogicTaskList.java

```diff
@@ -0,0 +1,151 @@
+package vimification.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+import vimification.model.task.Task;
+
+/**
+ * Responsible for storing, retrieving and updating all the tasks that are currently on the list.
+ */
+public class LogicTaskList {
+
+    private final List<Task> tasks;
+
+    public LogicTaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public LogicTaskList(List<Task> list) {
+        this.tasks = list;
+    }
+
+    public LogicTaskList(Task[] arr) {
+        this.tasks = Arrays.asList(arr);
+    }
+
+    public int length() {
+        return tasks.size();
+    }
+
+    //// Task-level operations
+
+    /**
+     * Returns the task with the specified index.
+     */
+    public Task get(int idx) {
+        return tasks.get(idx);
+    }
+
+    /**
+     * Returns true if a task that is the same as {@code t} exists in the task list.
+     */
+    public boolean has(Task t) {
+        requireNonNull(t);
+        for (Task task : tasks) {
+            if (t.isSameTask(task)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Adds a task to the task list.
+     */
+    public void add(Task t) {
+        requireNonNull(t);
+        tasks.add(t);
+    }
+
+    /**
+     * Inserts a task to the task list at the specified index.
+     */
+    public void insert(int idx, Task t) {
+        requireNonNull(t);
+        tasks.add(idx, t);
+    }
+
+    /**
+     * Removes the task with the specified index from the task list.
+     */
+    public Task delete(int idx) throws IndexOutOfBoundsException {
+        return tasks.remove(idx);
+    }
+
+    /**
+     * Removes the last task from the task list.
+     */
+    public void pop() {
+        delete(length() - 1);
+    }
+
+    /**
+     * Marks the task with the specified index as done.
+     */
+    public void markTask(int idx) {
+        get(idx).mark();
+    }
```

Comment:
For this one, just get the task out of the list using `get()` and call `mark()` on that one.  (Trust me, it will be much cleaner).